src/virtual_machine/register.rs

```diff
@@ -48,6 +48,14 @@ impl Registers {
             self.update_register(Register::ConditionFlag, Flag::POSITIVE as u16)
         }
     }
+
+    pub fn state_of_registers(&mut self) -> String {
+        self.registers
+            .iter()
+            .fold(String::new(), |accumulator, register| {
+                accumulator + &format!("{register:#018b}::")
+            })
+    }
```

Comment:
If I understand correctly, you are using `state_of_registers` as a way to serialize/format the registers, to be displayed in the debugger, right?

I think it would be much more idiomatic to `impl Display for Register`. That way, you get the `.to_string()` method for sending to the debugger.

Reply:
That makes sense, it's easier to understand that way